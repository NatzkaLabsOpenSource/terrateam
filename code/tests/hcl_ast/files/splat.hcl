foo = var.list[*].id
bar = var.list[*].interfaces[0].name
baz = var.*.zoom
