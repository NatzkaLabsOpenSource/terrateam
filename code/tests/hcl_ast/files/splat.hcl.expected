[
  [
    "Attribute",
    "foo",
    [
      "Attr",
      [
        [
          "Idx",
          [
            [ "Attr", [ [ "Id", "var" ], [ "A_string", "list" ] ] ],
            [ "Splat" ]
          ]
        ],
        [ "A_string", "id" ]
      ]
    ]
  ],
  [
    "Attribute",
    "bar",
    [
      "Attr",
      [
        [
          "Idx",
          [
            [
              "Attr",
              [
                [
                  "Idx",
                  [
                    [ "Attr", [ [ "Id", "var" ], [ "A_string", "list" ] ] ],
                    [ "Splat" ]
                  ]
                ],
                [ "A_string", "interfaces" ]
              ]
            ],
            [ "Int", 0 ]
          ]
        ],
        [ "A_string", "name" ]
      ]
    ]
  ],
  [
    "Attribute",
    "baz",
    [
      "Attr",
      [
        [ "Attr", [ [ "Id", "var" ], [ "A_splat" ] ] ],
        [ "A_string", "zoom" ]
      ]
    ]
  ]
]
